src/main/scala/ch/epfl/pop/crypto/Hash.scala

```diff
@@ -15,10 +16,8 @@ object Hash {
    * @return the id of the message
    */
   def computeMessageId(msg: Base64String, signature: Signature): Hash = {
-    val md = getMessageDigest()
-    md.update(msg.getBytes(StandardCharsets.UTF_8))
-    md.update(signature)
-    val id = md.digest()
+    val data = arrayRepr(msg, Base64.getEncoder.encodeToString(signature))
```

Comment:
You could also store the reference of the encoder/decoder created using `Base64.getEncoder` and `Base64.getDecoder`. 
As you can see in the [Java doc](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html), the creation of such object is a static method, but any encoding/decoding using `encodeToString(_)` for example is not static. This means that another instance of the object is created everytime

Reply:
So the concern is that we are creating a new object each time, right? Actually, if you look at source code, you can see that the same object is returned by every call to getEncoder. However, I'm not sure if it is a good idea to rely on this as it is internal detail and is subject to change.